### DISCUSSION TITLE
Mac or Wine (Linux)

### DISCUSSION THREAD

**Discussion by mlewus** [terjeio/ioSender#111] [Ideas]
I expect a Mac port is unlikely. Rather than a native port, has anyone tried to use ioSender with Wine on Mac or Linux?

**Comment by wolfgangr** [comment]
just tried it on debian - no success.

In a wineprefix with dotnet452, it keeps throwing until stack overflow:
```wine: Call from 0x7bc53069 to unimplemented function MSVCR100_CLR0400.dll._except_handler4_common, aborting```

In a wineprefix with dotnet35sp1 it says skd like "unable to find a version of the runtime"

**Comment by cbaugher** [comment]
I messed with IOSender and Wine a fair amount with no success.

Since I'm still interested in getting IOSender running on Linux I started working on a direct port of it last winter.  I made a fair amount of progress but there is still a ways to go before it's generally usable.  This winter I plan on starting work on it again, so hopefully there will be something to play with eventually.

**Reply by phil-barrett** [reply]
Sounds good.  I will be happy to test. Especially interested in it running on a RasPi

**Reply by cbaugher** [reply]
Yes, that is exactly my use case.  A pi based lathe and milling machine controller.

Honestly I could really use some help with the development since I'm not a Windows developer.  I've been doing embedded stuff for a good while but C#/GIU programming is a completely different thing.  I have a hard time wrapping my head around the concepts sometimes.

**Reply by phil-barrett** [reply]
There are several other threads in this section about porting to Linux.  You might find several kindred souls to help you.  It's not an easy task but definitely worthwhile.

**Comment by Switch-Witch** [comment]
I use IOSender on my Mac via Parallels wich works fine

**Comment by gerthalberg** [comment]
Has any of you noticed the recent activities regarding .Net on Linux from coming from MS? Like: https://learn.microsoft.com/en-us/dotnet/core/install/linux-ubuntu-install?tabs=dotnet9&pivots=os-linux-ubuntu-2410
